resource/name_test.go

```diff
@@ -0,0 +1,148 @@
+package resource
```

Comment:
I added 2 new functions to this package so I added some tests add added tests for the most common ways resource.Name s are created